test/end2end_test.go

```diff
@@ -5284,6 +5284,53 @@ func (s) TestGRPCMethod(t *testing.T) {
 	}
 }
 
+// renameProtoCodec is an encoding.Codec wrapper that allows customizing the
+// Name() of another codec.
+type renameProtoCodec struct {
+	encoding.Codec
+	name string
+}
+
+func (r *renameProtoCodec) Name() string { return r.name }
+
+func (s) TestForceCodecName(t *testing.T) {
+	wantContentTypeCh := make(chan string, 1)
+	defer close(wantContentTypeCh)
+
+	ss := &stubserver.StubServer{
+		EmptyCallF: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+			md, ok := metadata.FromIncomingContext(ctx)
+			if !ok {
+				return nil, status.Errorf(codes.Internal, "no metadata in context")
+			}
+			if got, want := md["content-type"], <-wantContentTypeCh; len(got) != 1 || got[0] != want {
```

Comment:
Nit: I don't like this much logic on a single line of code. However, this seems to be very consistent with go style and previous discussions I've had with the team (making a lot of conditional validation checks less verbose where verbose means lots of lines).

Reply:
Yeah I like that better.